restli-common/src/main/java/com/linkedin/restli/common/TypeSpec.java

```diff
@@ -67,7 +68,11 @@ public TypeSpec(Class<T> type)
   private static DataSchema backfillSchemaIfPossible(Class<?> type)
   {
     // These are all the classes used for type specs that are "schema-less".
-    if(type == CompoundKey.class || type == ComplexResourceKey.class || type == Void.class) return null;
+    if (type == CompoundKey.class || type == ComplexResourceKey.class || type == Void.class
+        || (!DataSchemaUtil.isPrimitiveClass(type) && DataTemplateUtil.hasCoercer(type)))
```

Comment:
Why don't you just keep track of registered custom classes in `Custom.java`? I think that would be simpler

Reply:
we could do that but didn't see much value for adding another map when this can be achieved using existing ones. Also, since coercers are only for customs, so felt like doing it this way. Let me know if you see any issue with this.

Reply:
true, though I feel like this can be more abstracted. What if you created a new helper method (e.g. `DataTemplateUtil#isCustomType(Class<?>)` which itself checks the coercers and does the logic to exclude primitives. It doesn't make sense to me that this consumer method needs to know that primitives also have coercers, it just wants to know if a type is a custom typeref/custom type/whatever.

Reply:
Another alternative is to create the helper method as `Custom#isCustomType(Class<?>)`, since it may not be appropriate for `DataTemplateUtil` to have a sense of what is "custom" and what isn't.

Reply:
done